apps/smart-contracts/core/test/WithdrawHook.test.ts

```diff
@@ -134,6 +144,157 @@ describe('=> WithdrawHook', () => {
 
       expect(depositRecord.recordWithdrawal).to.be.calledWith(user.address, TEST_AMOUNT_AFTER_FEE)
     })
+
+    describe('global withdraw limit testing', () => {
+      it('sets last global reset to current time if 0', async () => {
+        expect(await withdrawHook.getLastGlobalPeriodReset()).to.eq(0)
+
+        await withdrawHook
+          .connect(vault)
+          .hook(user.address, TEST_AMOUNT_BEFORE_FEE, TEST_AMOUNT_AFTER_FEE)
+
+        expect(await withdrawHook.getLastGlobalPeriodReset()).to.eq(
+          await getLastTimestamp(ethers.provider)
+        )
+      })
+
+      it('sets last global reset to current time if global period has passed (>)', async () => {
+        await withdrawHook
+          .connect(vault)
+          .hook(user.address, TEST_AMOUNT_BEFORE_FEE, TEST_AMOUNT_AFTER_FEE)
+        const previousResetTimestamp = await getLastTimestamp(ethers.provider)
+        expect(await withdrawHook.getLastGlobalPeriodReset()).to.eq(previousResetTimestamp)
+        await setNextTimestamp(
+          ethers.provider,
+          previousResetTimestamp + TEST_GLOBAL_PERIOD_LENGTH + 1
+        )
+
+        await withdrawHook
+          .connect(vault)
+          .hook(user.address, TEST_AMOUNT_BEFORE_FEE, TEST_AMOUNT_AFTER_FEE)
+
+        const currentResetTimestamp = await getLastTimestamp(ethers.provider)
+        expect(await withdrawHook.getLastGlobalPeriodReset()).to.eq(currentResetTimestamp)
+      })
+
+      it('sets global amount withdrawn to current amount being withdrawn if global period has passed (>)', async () => {
+        await withdrawHook
+          .connect(vault)
+          .hook(user.address, TEST_AMOUNT_BEFORE_FEE, TEST_AMOUNT_AFTER_FEE)
+        const differentAmountToWithdraw = 1
+        expect(await withdrawHook.getGlobalAmountWithdrawnThisPeriod()).to.not.eq(
+          differentAmountToWithdraw
+        )
+        const previousResetTimestamp = await getLastTimestamp(ethers.provider)
+        await setNextTimestamp(
+          ethers.provider,
+          previousResetTimestamp + TEST_GLOBAL_PERIOD_LENGTH + 1
+        )
+
+        await withdrawHook
+          .connect(vault)
+          .hook(user.address, differentAmountToWithdraw, differentAmountToWithdraw)
+
+        expect(await withdrawHook.getGlobalAmountWithdrawnThisPeriod()).to.eq(
+          differentAmountToWithdraw
+        )
+      })
+
+      it("doesn't update last global reset if global period was just reached (==)", async () => {
```

Comment:
nit:
```suggestion
      it("doesn't update last global reset if global period exactly reached", async () => {
```

similarly for others